src/lib/util/collapseHourlyMetrics.ts

```diff
@@ -21,31 +10,24 @@ const createMetricKey = (metric: IClientMetricsEnv): string => {
     ].join();
 };
 
-const sumYesNo = (
-    metrics: IClientMetricsEnv[],
-): Pick<IClientMetricsEnv, 'yes' | 'no'> => {
-    return metrics.reduce(
-        (acc, metric) => ({
-            yes: acc.yes + metric.yes,
-            no: acc.no + metric.no,
-        }),
-        {
-            yes: 0,
-            no: 0,
-        },
-    );
-};
-
 export const collapseHourlyMetrics = (
     metrics: IClientMetricsEnv[],
 ): IClientMetricsEnv[] => {
-    const hourlyMetrics = metrics.map((metric) => ({
-        ...metric,
-        timestamp: startOfHour(metric.timestamp),
-    }));
-
-    return groupBy(hourlyMetrics, createMetricKey).flatMap((group) => ({
-        ...group[0],
-        ...sumYesNo(group),
-    }));
+    const grouped = new Map<string, IClientMetricsEnv>();
+    metrics.forEach((metric) => {
+        const hourlyMetric = {
+            ...metric,
+            timestamp: startOfHour(metric.timestamp),
+        };
+        const key = createMetricKey(hourlyMetric);
+        if (!grouped[key]) {
+            grouped[key] = {
+                ...hourlyMetric,
+            };
```

Comment:
Nice work! I don't think this has to be a copy, since we just coped `metric` into `hourlyMetric`.

```ts
grouped[key] = hourlyMetric;
```

Reply:
https://github.com/Unleash/unleash/pull/1945/commits/12207c63871c41bc11c7be6e316a276a6a2940ad